packages/host/app/components/card-catalog/modal.gts

```diff
@@ -288,9 +308,13 @@ export default class CardCatalogModal extends Component<Signature> {
 
   @action
   setSearchKey(searchKey: string) {
-    this.searchKey = searchKey;
-    if (!this.searchKey) {
-      this.resetState();
+    if (!this.state) {
+      return;
+    }
+
+    this.state.searchKey = searchKey;
+    if (!this.state.searchKey) {
+      this.resetstateState();
```

Comment:
camel-case methods please

Reply:
also, this is really awkwardly named. please name this better

Reply:
Yep, it's a typo I think the name doesn't make sense. Updated [2281a83](https://github.com/cardstack/boxel/pull/616/commits/2281a83db15dd12a30e243f1082ca01539c0867a)